stellargraph/mapper/sampled_link_generators.py

```diff
@@ -223,6 +221,9 @@ def __init__(self, G, batch_size, num_samples, seed=None, name=None):
         self.num_samples = num_samples
         self.name = name
 
+        # Check if the graph has features
+        G.check_graph_for_ml()
```

Comment:
Hmm, maybe instead of having to add this line everywhere, we can just update `BatchedLinkGenerator.__init__` to be:
```python
class BatchedLinkGenerator(abc.ABC):
    def __init__(self, G, batch_size, schema=None, use_node_features=True):
        ...
        if use_node_features:
            G.check_graph_for_ml()

class Node2VecLinkGenerator(BatchedLinkGenerator):
    def __init__(self, G, batch_size, name=None):
        super().__init__(G, batch_size, use_node_features=False)
        ...
```
Would that work?

And similarly for the node generators

Reply:
Good idea!